tests/qos/test_pfc_pause.py

```diff
@@ -26,45 +29,90 @@
 """ Maximum number of interfaces to test on a DUT """
 MAX_TEST_INTFS_COUNT = 4
 
-def run_test_t0(fanouthosts,
-                duthost,
-                ptfhost,
-                conn_graph_facts,
-                leaf_fanouts,
-                dscp,
-                dscp_bg,
-                queue_paused,
-                send_pause,
-                pfc_pause,
-                pause_prio,
-                pause_time=65535,
-                max_test_intfs_count=128):
+@pytest.fixture(scope="module", autouse=True)
+def pfc_test_setup(duthosts, rand_one_dut_hostname):
     """
-    @Summary: Run a series of tests on a T0 topology.
-    For the T0 topology, we only test Vlan (server-faced) interfaces.
-    @param conn_graph_facts: Testbed topology
-    @param leaf_fanouts: Leaf fanout switches
-    @param dscp: DSCP value of test data packets
-    @param dscp_bg: DSCP value of background data packets
-    @param queue_paused: if the queue is expected to be paused
-    @param send_pause: send pause frames or not
-    @param pfc_pause: send PFC pause frames or not
-    @param pause_prio: priority of PFC franme
-    @param pause_time: pause time quanta. It is 65535 (maximum pause time quanta) by default.
-    @param max_test_intfs_count: maximum count of interfaces to test. By default, it is a very large value to cover all the interfaces.
-    return: Return # of iterations and # of passed iterations for each tested interface.
+    Generate configurations for the tests
+
+    Args:
+        duthosts(AnsibleHost) : multi dut instance
+        rand_one_dut_hostname(string) : one of the dut instances from the multi dut
+
+    Yields:
+        setup(dict): DUT interfaces, PTF interfaces, PTF IP addresses, and PTF MAC addresses
     """
 
-    """ Clear DUT's PFC counters """
-    duthost.sonic_pfc_counters(method="clear")
+    """ Get all the active physical interfaces enslaved to the Vlan """
+    """ These interfaces are actually server-faced interfaces at T0 """
+    duthost = duthosts[rand_one_dut_hostname]
+    vlan_members = get_active_vlan_members(duthost)
+
+    """ Get Vlan subnet """
+    vlan_subnet = get_vlan_subnet(duthost)
+
+    """ Generate IP addresses for servers in the Vlan """
+    vlan_ip_addrs = get_addrs_in_subnet(vlan_subnet, len(vlan_members))
+
+    """ Generate MAC addresses 00:00:00:00:00:XX for servers in the Vlan """
+    vlan_mac_addrs = [5 * '00:' + format(k, '02x') for k in random.sample(range(1, 256), len(vlan_members))]
+
+    """ Find correspoinding interfaces on PTF """
+    phy_intfs = get_phy_intfs(duthost)
+    phy_intfs.sort(key=natural_keys)
+    vlan_members.sort(key=natural_keys)
+    vlan_members_index = [phy_intfs.index(intf) for intf in vlan_members]
+    ptf_intfs = ['eth' + str(i) for i in vlan_members_index]
```

Comment:
Better to lookup from the ptf indices map so that it will work for all testbeds.

Reply:
Will address this separately. Need a bit more rework